client/selfrepair/tests/test_utils.js

```diff
@@ -125,18 +127,6 @@ describe('bucketSample', () => {
           const start = k === 0 ? 0 : bucketPoints[k - 1];
           const count = bucketPoints[k] - start;
```

Comment:
I think you should change this to

```js
for let k = 1; k < bucketPoints.length; k++) {
  const start = bucketPoints[k - 1];
  const count = bucketPoints[k] - start;
  // ...
}
```

since there is the explicit 0 at the beginning now.